pom.xml

```diff
@@ -56,8 +58,109 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>ssh-slaves</artifactId>
-            <version>1.25</version>
+            <version>1.27</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>structs</artifactId>
+            <version>1.17</version>
             <scope>test</scope>
         </dependency>
     </dependencies>
+    <profiles>
+        <profile>
+            <id>docker-generate-binaries</id>
+            <activation>
+                <os>
+                    <family>unix</family>
+                </os>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>exec-maven-plugin</artifactId>
+                        <version>1.6.0</version>
+                        <configuration>
+                            <executable>docker</executable>
+                            <workingDirectory>${binary.src}</workingDirectory>
+                        </configuration>
+                        <executions>
+                            <execution>
+                                <id>docker-build-image</id>
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>exec</goal>
+                                </goals>
+                                <configuration>
+                                    <arguments>
+                                        <argument>build</argument>
+                                        <argument>-t</argument>
+                                        <argument>${binary.dockerimage}:${revision}</argument>
+                                        <argument>.</argument>
+                                    </arguments>
+                                </configuration>
+                            </execution>
+                            <execution>
+                                <id>docker-generate-binaries</id>
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>exec</goal>
+                                </goals>
+                                <configuration>
+                                    <arguments>
+                                        <argument>run</argument>
+                                        <argument>-i</argument>
+                                        <argument>--rm</argument>
+                                        <argument>--mount</argument>
+                                        <argument>type=bind,src=${basedir}/${binary.src},dst=/org/jenkinsci/plugins/durabletask</argument>
```

Comment:
Better to take all the `docker` commands and put them in a separate shell script or something, so that the `maven-exec-plugin` configuration is short.

Reply:
will do